### DISCUSSION TITLE
Deprecation of nvim-lsp-installer.installers.* modules

### DISCUSSION THREAD

**Discussion by williamboman** [williamboman/nvim-lsp-installer#619] [Announcements]
I've done some significant overhauls recently, where as a result the modules within `nvim-lsp-installer.installers.*` are now deprecated and should not be used. I've written a short doc describing the changes here: https://github.com/williamboman/nvim-lsp-installer/wiki/Async-infrastructure-changes-notice. For most use cases, the changes needed should be rather minimal.

**Comment by mracos** [comment]
Hey! Thx for the plugin and the care that you put into it 💙 

One question, I'm using the `require("nvim-lsp-installer.installers.shell").polyshell` to install a custom server, should I use `ctx.spawn.bash` or something else? One note is that all my setup is through `zsh`, would it pick up the default shell to run commands or would it use `bash`?

I ask because I need some stuff to install a server that is set up through `zsh`, which is `asdf`

**Reply by williamboman** [reply]
Hello! Unless you're running an actual shell script, there are actually not a lot of reasons to shell out to bash/zsh/sh etc. You could just spawn the processes directly (you could think of the async installer function as a shell interface, but in Lua). When spawning processes via `ctx.spawn` or `spawn` directly, it will just use the field name you provide as the command, so `ctx.spawn.bash {}` would try to spawn `bash`, `ctx.spawn["/bin/bash"] {}` would spawn `/bin/bash`, `ctx.spawn.zsh {}` would spawn `zsh` and so on.

If `asdf` is in your PATH in the shell in which you launch `nvim` (does `:echo executable("asdf")` yield `1`), you could get by by doing something like:

```lua
installer = function (ctx)
    ctx.spawn.asdf { "install", "something" }
end
```

**Reply by mracos** [reply]
I see, thanks for the context and for the help, it worked perfectly using the `ctx.spawn.$executable`.

For reference, I had:

```
shell.polyshell([[
  mix deps.get --force && \
    mix compile && \
    mix elixir_ls.release -o elixir-ls && \
    chmod +x elixir-ls/language_server.sh
]]),
```

Which I changed to:

```
ctx.spawn.mix({ "deps.get", "--force" })
ctx.spawn.mix({ "compile" })
ctx.spawn.mix({ "elixir_ls.release", "-o", "elixir-ls" })
ctx.spawn.chmod({ "+x", "elixir-ls/language_server.sh" })
```
